binder/src/test/java/io/grpc/binder/internal/ActiveTransportTrackerTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.binder.internal;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.truth.Truth.assertThat;
+
+import static org.junit.Assert.assertThrows;
+
+import android.os.Binder;
+import android.os.IBinder;
+import com.google.common.collect.ImmutableList;
+import io.grpc.Attributes;
+import io.grpc.Metadata;
+import io.grpc.internal.FixedObjectPool;
+import io.grpc.internal.ObjectPool;
+import io.grpc.internal.ServerListener;
+import io.grpc.internal.ServerStream;
+import io.grpc.internal.ServerTransport;
+import io.grpc.internal.ServerTransportListener;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+@RunWith(RobolectricTestRunner.class)
+public final class ActiveTransportTrackerTest {
+
+  private static final OneWayBinderProxy.Decorator NO_OP_BINDER_DECORATOR = input -> input;
+
+  private final ObjectPool<ScheduledExecutorService> executorServicePool =
+      new FixedObjectPool<>(Executors.newSingleThreadScheduledExecutor());
+  private final IBinder callbackBinder = new Binder();
+
+  private boolean serverShutdownNotified = false;
+  private final Runnable testShutdownListener = () -> serverShutdownNotified = true;
+  private ActiveTransportTracker tracker;
+
+  @Before
+  public void setUp() {
+    tracker = new ActiveTransportTracker(new NoOpServerListener(), testShutdownListener);
```

Comment:
Can you add test cases that ensure the methods of io.grpc.internal.ServerListener are forwarded to the delegate?Maybe use a mock instead of this noop impl?

Reply:
Added.